zenmap/zenmapGUI/ScanInterface.py

```diff
@@ -429,33 +430,46 @@ def start_scan_cb(self, widget=None):
         command = self.command_toolbar.command
         profile = self.toolbar.selected_profile
 
-        log.debug(">>> Start Scan:")
-        log.debug(">>> Target: '%s'" % target)
-        log.debug(">>> Profile: '%s'" % profile)
-        log.debug(">>> Command: '%s'" % command)
+        check = subprocess.Popen(["mktemp"], stdout=subprocess.PIPE).communicate()[0]
 
-        if target != '':
-            try:
-                self.toolbar.add_new_target(target)
-            except IOError, e:
-                # We failed to save target_list.txt; treat it as read-only.
-                # Probably it's owned by root and this is a normal user.
-                log.debug(">>> Error saving %s: %s" % (
-                    Path.target_list, str(e)))
-
-        if command == '':
+        if (not check):
```

Comment:
Instead of a `if/else` check here, probably use a `try/except`. This will let us catch the exception and report the error text directly to the user, which contains the list of directories tried.